Applications/Spire/Source/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,123 @@
+#include "Spire/OrderImbalanceIndicator/FilteredOrderImbalanceIndicatorModel.hpp"
+#include "Nexus/Definitions/Market.hpp"
+#include "Spire/Spire/QtPromise.hpp"
+
+using Filter = Spire::FilteredOrderImbalanceIndicatorModel::Filter;
+using namespace Nexus;
+using namespace Spire;
+
+FilteredOrderImbalanceIndicatorModel::FilteredOrderImbalanceIndicatorModel(
+    std::shared_ptr<OrderImbalanceIndicatorModel> source_model,
+    std::vector<Filter> filters)
+    : m_source_model(std::move(source_model)),
+      m_filters(std::move(filters)) {}
+
+OrderImbalanceIndicatorModel::SubscriptionResult
+    FilteredOrderImbalanceIndicatorModel::subscribe(
+    const boost::posix_time::ptime& start,
+    const boost::posix_time::ptime& end,
+    const OrderImbalanceSignal::slot_type& slot) {
+  auto subscription = std::make_shared<Subscription>(start, end);
```

Comment:
Eliminate as much as possible and reuse the existing 1-1 relationship between the ```FilteredOrderImbalanceIndicatorModel``` and the ```m_source_model```.